File: ICPC_Preli_2024/E.cpp
#include <iostream>
using namespace std;

int main() {
    int T;
    cin >> T;

    for (int i = 1; i <= T; ++i) {
        long long N;
        int P;
        cin >> N >> P;

        string winner;
        if (N % 2 == 1) {
            winner = (P == 1) ? "Evenius" : "Oddius";
        } else { 
            winner = (P == 2) ? "Oddius" : "Evenius";
        }

        cout << "Case " << i << ": " << winner << endl;
    }

    return 0;
}

File: ICPC_Preli_2024/E.py
import sys
input = sys.stdin.read

def main():
    data = input().strip().split()
    T = int(data[0])
    results = []

    for i in range(1, T + 1):
        N = int(data[2 * i - 1])
        P = int(data[2 * i])

        if N % 2 == 1:  # Odd number of stones
            winner = "Oddius" if P == 2 else "Evenius"
        else:  # Even number of stones
            winner = "Evenius" if P == 2 else "Oddius"

        results.append(f"Case {i}: {winner}")

    print("\n".join(results))

main()
